src/pg/mod.rs

```diff
@@ -102,6 +103,7 @@ pub struct AsyncPgConnection {
     stmt_cache: Arc<Mutex<StmtCache<diesel::pg::Pg, Statement>>>,
     transaction_state: Arc<Mutex<AnsiTransactionManager>>,
     metadata_cache: Arc<Mutex<PgMetadataCache>>,
+    error_receiver: Arc<Mutex<tokio::sync::oneshot::Receiver<diesel::result::Error>>>,
```

Comment:
I'm not sure if using a mutex here is a good idea, because that one would be locked while executing the query (compared to while starting to prepare as for the other mutexes). That would basically restrict the ability to do pipe lining at all. I need to think about a better solution, probably using a `mpmc` channel or something like that.

Reply:
Perhaps we could make it only take the mutex as part of the select arm so it doesn't lock the whole time. Not sure if that works though.

Reply:
Hmm actually that doesn't really work yeah, you need to take the lock to await the channel as a future. Perhaps some approach where we use try_recv and if it returns Ok, we return a completed future. I tried to code this up but it didn't really work.

I notice that some other tokio channels have maybe what we need, this poll_recv function: https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Receiver.html#method.poll_recv.

Reply:
I've finally found the time to have another look at this. #132 is what I've settled on.

Reply:
Neato, looks good to me! I'll close this one out.